plugins/extractors/googleworkspace/googleworkspace.go

```diff
@@ -0,0 +1,218 @@
+    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+    "client_x509_cert_url": "XXXXXX"
+}
+user_email: user@odpf.com`
+
+var info = plugins.Info{
+	Description:  "User list from Google Workspace",
+	SampleConfig: sampleConfig,
+	Tags:         []string{"platform", "extractor"},
+	Summary:      summary,
+}
+
+// Extractor manages the extraction of data from the extractor
+type Extractor struct {
+	plugins.BaseExtractor
+	logger      log.Logger
+	config      Config
+	TokenSource oauth2.TokenSource
+	Client      *admin.Service
+	emit        plugins.Emit
+}
+
+// New returns a pointer to an initialized Extractor Object
+func New(logger log.Logger) *Extractor {
+	e := &Extractor{
+		logger: logger,
+	}
+	e.BaseExtractor = plugins.NewBaseExtractor(info, &e.config)
+
+	return e
+}
+
+// Init initializes the extractor
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) (err error) {
+	if err = e.BaseExtractor.Init(ctx, config); err != nil {
+		return err
+	}
+
+	jwtConfig, err := google.JWTConfigFromJSON([]byte(e.config.ServiceAccountJSON), admin.AdminDirectoryUserScope)
+	if err != nil {
+		return plugins.InvalidConfigError{}
+	}
+
+	if e.config.UserEmail == "" {
+		return plugins.InvalidConfigError{}
+	}
+	jwtConfig.Subject = e.config.UserEmail
+	ts := jwtConfig.TokenSource(ctx)
+
+	e.TokenSource = ts
+
+	return
+}
+
+// Extract extracts the data from the extractor
+// The data is returned as a list of assets.Asset
+func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error) {
+	var status string
+	e.emit = emit
+	r, err := FetchUsers(ctx, e.TokenSource)
+	if err != nil {
+		return err
+	}
+
+	if len(r.Users) == 0 {
+		e.logger.Info("No users found.\n")
+		return nil
+	}
+
+	for _, u := range r.Users {
+		if !u.Suspended {
+			status = "not suspended"
+		} else {
+			status = "suspended"
+		}
+
+		var userAttributes = make(map[string]interface{})
+		userAttributes = buildOrgAttributes(userAttributes, u.Organizations)
+		userAttributes = buildRelationsAttributes(userAttributes, u.Relations)
+		userAttributes = buildCustomSchemasAttributes(userAttributes, u.CustomSchemas)
+		userAttributes["org_unit_path"] = u.OrgUnitPath
+		userAttributes["aliases"] = strings.Join(u.Aliases, ",")
+
+		user, err := anypb.New(&assetsv1beta2.User{
+			Email:      u.PrimaryEmail,
+			FullName:   u.Name.FullName,
+			Status:     status,
+			Attributes: utils.TryParseMapToProto(userAttributes),
+		})
+		if err != nil {
+			return err
+		}
+
+		e.emit(models.NewRecord(&assetsv1beta2.Asset{
+			Data: user,
+		}))
+	}
+
+	return nil
+}
+
+// init registers the extractor to catalog
+func init() {
+	if err := registry.Extractors.Register("googleworkspace", func() plugins.Extractor {
+		return New(plugins.GetLog())
+	}); err != nil {
+		panic(err)
+	}
+}
+
+func FetchUsers(ctx context.Context, ts oauth2.TokenSource) (*admin.Users, error) {
+	srv, err := admin.NewService(ctx, option.WithTokenSource(ts))
```

Comment:
Do we need to build service every time? or can we just do it once on `Init()` method?